sites/game-client/ui/overlay.mjs

```diff
@@ -1,113 +1,158 @@
 // Copyright 2021 HITCON Online Contributors
 // SPDX-License-Identifier: BSD-2-Clause
 
-// enum
-const OverlayPosition = Object.freeze({
-    LEFT_TOP: Symbol(),
-    LEFT_CENTER: Symbol(),
-    LEFT_BOTTOM: Symbol(),
-    RIGHT: Symbol()
-    //CENTER_TOP: Symbol(), // overlapped with notification
-    /* 
-    CENTER_BOTTOM: Symbol(),
-    RIGHT_TOP: Symbol(),
-    RIGHT_CENTER: Symbol(),
-    RIGHT_BOTTOM: Symbol()
-    */
-});
-
 /** 
  * Represents an overlay
  * 
  * Those who wants to use the Overlay element should extend this class and implement the methods.
 */
 class Overlay {
-    /**
-     * 
-     * @param dom HTML DOM element.
-     */
-    constructor(dom) {}
-  
-    // Overlay hide/show related properties.
-    /**
-     * If true, user can hide the overlay.
-     * 
-     * This property should be implemented by the base class.
-     */
-    canHide() {}
-
-    /**
-     * If true, user can dismiss the overlay.
-     *
-     * This property should be implemented by the base class.
-     */
-    canDismiss() {}
-    
-    /**
-     * Unset ToolbarButton display=none.
-     */
-    showToolbarButton() { }
-
-    /**
-     * Set ToolbarButton display=none.
-     */
-    hideToolbarButton() { }
-
-    /**
-     * If the overlay has ToolbarButton, it should toggle the overlay hide/show state
-     * when the button is clicked.
-    */
-    onClickToolbarButton()
+  /**
+   * @param dom HTML DOM element.
+   */
+  constructor(mainUI, dom) {
+    this.mainUI = mainUI;
+    this.dom = dom;
+    this.position = undefined;
+  }
   
-    /*
-    -> HIDDEN -> SHOW -+
-       |  ^-------|    |
-       +---------------+
-       V
-    DISMISSED
-    */
+  /**
+   * If true, user can hide the overlay.
+   * 
+   * This property should be implemented by the derived class.
```

Comment:
Consider adding `console.error(function_name + 'should be implemented in derived class!')` to ensure the implementation.

Reply:
The comment is the one that's incorrect. I've adjusted the comment to reflect the fact that implementing that method is optional. i.e. "can" vs "should".